src/Plugin/Field/FieldWidget/ExploreCoursesUrlWidget.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+namespace Drupal\stanford_courses\Plugin\Field\FieldWidget;
+
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\link\Plugin\Field\FieldWidget\LinkWidget;
+use GuzzleHttp\ClientInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the 'explore_courses_url' widget.
+ *
+ * @FieldWidget(
+ *   id = "explore_courses_url",
+ *   label = @Translation("ExploreCourses"),
+ *   field_types = {
+ *     "link"
+ *   }
+ * )
+ */
+class ExploreCoursesUrlWidget extends LinkWidget {
+
+  /**
+   * Http Client Service.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $client;
+
+  /**
+   * {@inheritDoc}.
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['third_party_settings'],
+      $container->get('http_client')
+    );
+  }
+
+  /**
+   * {@inheritDoc}.
+   */
+  public static function defaultSettings() {
+    $settings = [
+      'api_version' => '',
```

Comment:
Should be some default setting, even if it's the current version. I've seen where you can set the widget and then forget to set the widget settings. Then things don't work correctly when using the widget.

Reply:
default added.